compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -664,7 +664,7 @@ fn codegen_regular_intrinsic_call<'tcx>(
                 return;
             }
 
-            if intrinsic == sym::assert_zero_valid && !layout.might_permit_raw_init(fx, /*zero:*/ true) {
+            if intrinsic == sym::assert_zero_valid && !layout.might_permit_raw_init(fx, /*zero:*/ true, fx.tcx.sess.opts.debugging_opts.strict_init_checks) {
```

Comment:
Would you mind formatting this such that the line length stays below 100 characters? This is inside a macro, so rustfmt doesn't pick up on it. Alternatively I can do this myself after this PR gets merged. Not a big deal.

Reply:
Btw, does cranelift not use the codegen_ssa crate? I was surprised to see the logic duplicated here.

Reply:
Ah, my bad, didn't spot that, sorry!

Bors was already kicked off so I don't think I can change it now, but if bors comes back with an error I need to fix I'll format this code at the same time. Otherwise, best for you to do it.

Reply:
Also, @bjorn3, I was planning to change the `zero` bool parameter here to a `enum ValidAs { Zero, Uninit }` for clarity, seeing as we need to comment what it actually is at the moment, but that requires me to add a `use rustc_target::abi::ValidAs` to the rustc_codegen_cranelift code somewhere.

I see there's an `prelude` module, should I be adding it there or did you want me to just add it in the one file it's used (`intrinsics/mod.rs`)

Reply:
> Btw, does cranelift not use the codegen_ssa crate? I was surprised to see the logic duplicated here.

Only the linker and metadata code are shared at the moment. cg_ssa is too much tied to LLVM at the moment. I did some refactorings for this quite a while ago, but didn't really do much about it recently. cg_clif also has a couple of peephole optimizations during codegen that cg_ssa is missing. And finally cg_clif directly generates ssa form where possible, while cg_ssa depends on a mem2reg pass that cranelift is missing.

Reply:
> I've just put it in the top of the file where it's needed for now.

Makes sense. The prelude is just for things used in many modules. I should probably make it smaller.